pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -113,10 +120,17 @@ type GrpcPodConfig struct {
 	// +optional
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
 
-	// RunAsRoot allows admins to indicate that they wish to run the container inside the CatalogSource pod in a privileged
-	// mode as root.  This should only be enabled when running older catalog images which could not be run as non-root.
+	// SecurityContextConfig can be one of "legacy" or "restricted". The grpc pod created is either injected with
+	// the right pod.spec.securityContext and pod.spec.container[*].securityContext values to allow the pod to run in
+	// Pod Security Admission(PSA) "restricted" mode, or does not set these values at all, in which case the pod
+	// can only be run in PSA "privileged" namespaces. By default, SecurityContextConfig is set to "restricted". If the value
+	// is unspecified, the default value of restricted is used.  Specifying any other value will result in a validation error.
+	// When using older catalog images, which could not be run in restricted mode, the SecurityContextConfig should be set to
+	// "legacy". More information about PSA can be found here: https://kubernetes.io/docs/concepts/security/pod-security-admission/
```

Comment:
Documentation Nit, I felt that some of the information concerning possible values to be apparent through the schema validation changes Anik recently introduced. Can be ignored if you don't think value is added @anik120 
```suggestion
	// SecurityContextConfig allows users to configure a grpc pod's pod.spec.securityContext and pod.spec.container[*].securityContext so the pod may meed supported Pod Security Admission(PSA) modes.
	// By default, the grpc pod will be ran in "restricted" mode.
	// If set to "legacy" mode, the grpc pod's pod.spec.securityContext and pod.spec.container[*].securityContext
	// will not be set and the pod will only run in PSA "privileged" namespaces. In most cases, "legacy" mode
	// should be used with older catalog images, which could not be run in restricted mode, 
	// More information about PSA can be found here: 
	// https://kubernetes.io/docs/concepts/security/pod-security-admission/
```

Reply:
@awgreene changed the doc, with slight modifications. PTAL!

Reply:
LGTM